### PR TITLE
vault: add `jwt_backend_path` agent config

### PR DESCRIPTION
**Pull request by lgfa29**
Add agent configuration to allow cluster operators to define the path where the JWT auth method backend is mounted.

### CODE REVIEW

File: nomad/structs/config/vault.go [hunk 1]
```diff
@@ -61,6 +61,9 @@ type VaultConfig struct {
 	// URL such as "http://vault.example.com"
 	Addr string `mapstructure:"address"`
 
+	// JWTBackendPath is the path used to access the JWT auth method.
+	JWTBackendPath string `mapstructure:"jwt_backend_path"`
```
Comment (lgfa29) [inline nomad/structs/config/vault.go:65]: I'm open to suggestions on how to call this field. I followed the language from https://developer.hashicorp.com/vault/docs/auth/jwt#via-the-cli

> The default path for the JWT authentication backend is `/jwt`

Comment (tgross) [reply nomad/structs/config/vault.go:65]: Maybe `JWTAuthBackendPath`? I don't feel strongly about it though.

File: website/content/docs/configuration/vault.mdx [hunk 2]
```diff
@@ -97,6 +97,10 @@ agents with [`client.enabled`] set to `true`.
   given in the format `protocol://host:port`. If your Vault installation is
   behind a load balancer, this should be the address of the load balancer.
 
+- `jwt_backend_path` - `(string: "jwt")` - Specifies the mount
+  [path][vault_auth_enable_path] of the JWT authentication method to be used to
+  login with workload identity JWTs.
```
Comment (tgross) [inline website/content/docs/configuration/vault.mdx:102]: :+1: for linking to the Vault docs here.

File: command/agent/config_parse_test.go [hunk 3]
```diff
@@ -318,6 +319,7 @@ var basicConfig = &Config{
 		structs.VaultDefaultCluster: {
 			Name:                 structs.VaultDefaultCluster,
 			Addr:                 "127.0.0.1:9500",
+			JWTBackendPath:       "nomad_jwt",
```
Comment (tgross) [inline command/agent/config_parse_test.go:322]: There's a `TestConfig_MultipleVault` test case below where it'd be nice to add an assertion for this vs the default (I think both will show up with the config changes you've made here).

**Review by tgross — APPROVED** [review]
LGTM!

**Pull request merged**